Source/Kesmai.Server/Game/Items/Equipment/Weapons/Weapon.cs

```diff
@@ -59,9 +59,22 @@ public override int CalculateBlockingBonus(ItemEntity item)
 		/// </summary>
 		public override double CalculateFumbleChance(MobileEntity entity)
 		{
-			// Skill Level = 3 =>  ((3 + 1)^2) * 10 = 160 => 1 / 160;
-			// Skill Level = 4 =>  ((4 + 1)^2) * 10 = 250 => 1 / 250;
-			return 1 / (10 * Math.Pow(entity.GetSkillLevel(Skill) + 1, 2));
+			var leftHand = LeftHand;
+			var rightHand = RightHand;
+
+			var leftWeapon = leftHand as IWeapon;
+			var rightWeapon = rightHand as IWeapon;
+
+			if(leftWeapon != null && rightHand != null && rightWeapon.Flags.HasFlag(WeaponFlags.TwoHanded))
```

Comment:
I think this should be rightWeapon != null since you are only checking for the TwoHanded flag.  could potentially crash if right hand is not a weapon.  Not sure if we can use the Elvis operator, but that is an option as well.

Reply:
Sounds good, I'll submit a new one. I could do a try catch and bail out. I think that will catch the weapon portion plus adding another conditional. Just makes it so it steps over.